webapi/src/main/twirl/queries/sparql/v2/generateInsertStatementsForStandoffLinks.scala.txt

```diff
@@ -61,15 +65,19 @@
             If this template is being used to create a single text value containing standoff links, the new LinkValue
             we are creating may be a new version of an existing LinkValue (linkUpdate.linkValueExists will be true).
             In that case, the WHERE clause will have already bound a SPARQL variable ?linkValue@linkValueIndex,
-            containing the IRI of the existing LinkValue. In that case, add a triple indicating that the new LinkValue
-            is a new version of the existing one.
+            containing the IRI of the existing LinkValue. Therefore, add a triple indicating that the new LinkValue
+            is a new version of the existing one, and copy the UUID from the existing one.
 
             If this template is being used to create a new resource with its initial values, there won't be any existing
-            LinkValues for standoff links, so linkUpdate.linkValueExists will be false.
+            LinkValues for standoff links, so linkUpdate.linkValueExists will be false. Generate a random UUID for the
+            LinkValue.
 
             *@
             @if(linkUpdate.linkValueExists) {
-                <@linkUpdate.newLinkValueIri> knora-base:previousValue ?linkValue@linkValueIndex .
+                <@linkUpdate.newLinkValueIri> knora-base:previousValue ?linkValue@linkValueIndex ;
```

Comment:
Wouldn't you have to remove the UUID from the previous version?

Reply:
Yes, that happens here:

https://github.com/dhlab-basel/Knora/blob/00267e66a277786e9fbe25bfafe2f32b1d22ed99/webapi/src/main/twirl/queries/sparql/v1/createValue.scala.txt#L88

I want to add another test to make sure these SPARQL updates insert and delete the UUIDs correctly.

Reply:
> I want to add another test to make sure these SPARQL updates insert and delete the UUIDs correctly.

On this PR?

Reply:
Yes, I'll do it today.

Reply:
Great. In the meantime, I will test the update script.